spacy/pipeline/entityruler.py

```diff
@@ -22,7 +55,16 @@ class EntityRuler:
     USAGE: https://spacy.io/usage/rule-based-matching#entityruler
     """
 
-    def __init__(self, nlp, phrase_matcher_attr=None, validate=False, **cfg):
+    def __init__(
+        self,
+        nlp: Language,
+        name: str = "entity_ruler",
+        phrase_matcher_attr: Optional[Union[int, str]] = None,
+        validate: bool = False,
+        overwrite_ents: bool = False,
+        ent_id_sep: str = DEFAULT_ENT_ID_SEP,
+        patterns: Optional[List[PatternType]] = None,
+    ):
```

Comment:
we have the same set of default values here...

Reply:
Oh, this happens in the PR throughout. Like `incl_prior=True` for the `EntityLinker` is defined three times?

Reply:
I think it's reasonable to not use function defaults on factories – they won't be used and it communicates the wrong thing. So I'll remove them.

Not sure how we should go about default arguments on the `Pipe` subclasses etc. In theory, those could still be instantiated separately, and the factory arguments don't _have to_ match the subclass arguments. So I think it makes sense to keep them there, even if it causes some duplication. But I guess it's not too weird to have a class with default args and then a function that instantiates it and passes in args.